pkg/asset/machines/azure/machines.go

```diff
@@ -105,6 +105,23 @@ func provider(platform *azure.Platform, mpool *azure.MachinePool, osImage string
 		managedIdentity = ""
 	}
 
+	var diskEncryptionSet *machineapi.DiskEncryptionSetParameters
+	if mpool.OSDisk.DiskEncryptionSet != nil &&
+		mpool.OSDisk.DiskEncryptionSet.SubscriptionID != "" &&
+		mpool.OSDisk.DiskEncryptionSet.ResourceGroup != "" &&
+		mpool.OSDisk.DiskEncryptionSet.Name != "" {
```

Comment:
These empty checks shouldn't be required because the fields have already been validated or have a default

Reply:
Fixed.